### PR TITLE
Feat/metrics db

### PR DESCRIPTION
**Pull request by jgryffindor**
We are adding a local metrics table to Talib's database for seeding values from Prometheus on a schedule. This reduces the load on the Prometheus server directly and we have more control over the metrics we expose to the end users. 

## Description

- Added metric scheduler to pull metrics on an interval. The job will seed values into the local DB in batches for each metric starting at a start date stored in the Prometheus query table. 
- Added metric entity and controller GET endpoints to pull current and series metrics directly from the table 
- Added DELETE endpoint to delete metrics stored in the database for a given metric
- Removed Prometheus query endpoints that were previously pulling directly from Prometheus
- Refactored Talib metricChart and metricStat components to pull from the new metrics Series/Current endpoints

## Testing
- Changes were tested locally 
- Connection to the Prometheus server was severed and no service failures occurred, errors log during disconnect. 
- Malformed Prometheus queries were inserted without impacting the functionality of healthy metrics.

## Breaking Changes (if applicable)
- This adds a new column to the existing Prometheus query tables 
- A database backup will be taken before merging and Prometheus queries can be deleted and re-inserted if anything fails. 
- The DELETE endpoint allows us to reset any metric from any start date we set in the Prometheus query record

## Checklist:

- [x] I have read and followed the CONTRIBUTING guidelines for this project.
- [x] I have added or updated tests and they pass.
- [x] I have added or updated documentation and it is accurate.
- [x] I have noted any breaking changes in this module or downstream modules.

### CODE REVIEW

File: server/.env [hunk 1]
```diff
@@ -38,10 +38,32 @@ GRAFANA_API_URL="https://grafana.liftedinit.tech"
...
 # is required.
 ADMIN_USERNAME=admin
 ADMIN_PASSWORD=admin
+
+# ====  Metrics SCHEDULER ====
+# Scheduler cron job running. If the scheduler hasn't finished work before the
+# next job, it will skip the schedule time.
+# If this and METRICS_SCHEDULER_SECONDS is missing, scheduler is disabled.
+# If both are set, this is an error.
+METRICS_SCHEDULER_CRON="*/60 * * * * *"
```
Comment (fmorency) [inline server/.env:55]: Is that even a valid cron string? Shouldn't it be something like `*/59 * * * *`, i.e., valid values for seconds are 0-59 and there's an additional `*` in your string for some reason?

File: server/.env [hunk 2]
```diff
@@ -38,10 +38,32 @@ GRAFANA_API_URL="https://grafana.liftedinit.tech"
...
+
+# Maximum amount of metrics to fetch schedule.
+# Since this is blocking future jobs, this should be a reasonable amount
+# to fetch, as we also need to fetch the latest block to update the height
+# of the whole blockchain.
```
Comment (fmorency) [inline server/.env:60]: Doesn't seem to apply to the following variable

File: server/.env [hunk 3]
```diff
@@ -38,10 +38,32 @@ GRAFANA_API_URL="https://grafana.liftedinit.tech"
...
+METRICS_SCHEDULER_STARTDATE_TIMESTAMP=1684076400000
+# Minimum batch size, scheduler will not run if a metric has less than 
+# this amount of metric datapoints to collect
+METRICS_SCHEDULER_MIN_BATCH_SIZE=20
+# Max batch size for each scheduler run
+# Multiplied by the number of metrics
+METRICS_SCHEDULER_BATCH_SIZE=100
+# Interval in ms for prometheus queries
+METRICS_SCHEDULER_INTERVAL=600000
```
Comment (fmorency) [inline server/.env:69]: So 1 minute.

File: server/src/config/metrics-scheduler/configuration.module.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import * as Joi from "joi";
+import configuration from "./configuration";
+import { MetricsSchedulerConfigService } from "./configuration.service";
+
+const validationSchema = Joi.object({
+  METRICS_SCHEDULER_CRON: Joi.custom((value, helper) => {
+    new CronTime(value);
+    return value;
+  }),
+  METRICS_SCHEDULER_SECONDS: Joi.number(),
```
Comment (fmorency) [inline server/src/config/metrics-scheduler/configuration.module.ts:13]: Where is this defined?

**Review by fmorency — APPROVED** [review]
I very quickly glanced at the code but don't have the expertise to really perform an in-depth FE code review.

Some minor comments. Thanks!

File: server/.env (hunk 2 above)
Comment (jgryffindor) [reply server/.env:60]: Will fix this comment!

File: server/.env (hunk 3 above)
Comment (jgryffindor) [reply server/.env:69]: 10 min interval between metric datapoints

File: server/.env (hunk 1 above)
Comment (jgryffindor) [reply server/.env:55]: See the SCHEDULER_CRON value from above. I cloned the setup for Talib scheduler and agree it is a different format than traditional Linux cron.  It is parsing this a 1-minute increment, but I will test with a traditional cron format.

**Pull request merged**